### PR TITLE
Remove set expiry from blob client

### PR DESCRIPTION
**Pull request by souravgupta-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/internal/exported/set_expiry.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+func (e ExpiryTypeRelativeToNow) Format(o *SetExpiryOptions) (generated.ExpiryOptions, *generated.BlobClientSetExpiryOptions) {
+	return generated.ExpiryOptionsRelativeToNow, &generated.BlobClientSetExpiryOptions{
+		ExpiresOn: to.Ptr(strconv.FormatInt(time.Duration(e).Milliseconds(), 10)),
+	}
+}
+
+func (e ExpiryTypeRelativeToCreation) Format(o *SetExpiryOptions) (generated.ExpiryOptions, *generated.BlobClientSetExpiryOptions) {
+	return generated.ExpiryOptionsRelativeToCreation, &generated.BlobClientSetExpiryOptions{
+		ExpiresOn: to.Ptr(strconv.FormatInt(time.Duration(e).Milliseconds(), 10)),
```
Comment (vibhansa-msft) [inline sdk/storage/azblob/internal/exported/set_expiry.go:55]: How are we taking creation time into account here?

Comment (souravgupta-msft) [reply sdk/storage/azblob/internal/exported/set_expiry.go:55]: We are not taking creation time into account in the SDK code. It is something which the backend API does.
Please refer this testcase, [TestBlockBlobSetExpiryRelativeToCreation](https://github.com/Azure/azure-sdk-for-go/blob/sourav/setExpiry/sdk/storage/azblob/blockblob/client_test.go#L3684)

File: sdk/storage/azblob/appendblob/models.go [hunk 2]
```diff
@@ -164,3 +164,29 @@ func (o *SealOptions) format() (*generated.LeaseAccessConditions,
 }
 
 // ---------------------------------------------------------------------------------------------------------------------
+
+// ExpiryOptions defines values for ExpiryOptions
+type ExpiryOptions = generated.ExpiryOptions
```
Comment (jhendrixMSFT) [inline sdk/storage/azblob/appendblob/models.go:169]: Do we actually need to export this?  It was replaced by `ExpiryType` yes?

Comment (souravgupta-msft) [reply sdk/storage/azblob/appendblob/models.go:169]: `ExpiryOptions` is used by the following method. We can remove it if we want.
```
// PossibleExpiryOptionsValues returns the possible values for the ExpiryOptions const type.
func PossibleExpiryOptionsValues() []ExpiryOptions {
	return generated.PossibleExpiryOptionsValues()
}
```

Comment (jhendrixMSFT) [reply sdk/storage/azblob/appendblob/models.go:169]: Yes looks like this is remnants that can be removed.

Comment (souravgupta-msft) [reply sdk/storage/azblob/appendblob/models.go]: Done

**Review by jhendrixMSFT — APPROVED** [review]

**Review by vibhansa-msft — APPROVED** [review]

**Pull request merged**